simulator-example/src/simulation.cpp

```diff
@@ -7,7 +7,8 @@ int main(int argv, char* argc)
     CacheHierarchySimulator::Api api;
 
     // Create an l1 cache
-    CacheHierarchySimulator::Cache l1(64, 8192, 64, 4, 5);
+    CacheHierarchySimulator::Cache l1(64, 8192, 64, 4, 5, CacheHierarchySimulator::ReplacementPolicy::FIFO, CacheHierarchySimulator::WRITE_BACK_ALLOCATE);
```

Comment:
Are there plans to eventually get this information from a configuration file?

Reply:
We might could make the config file a JSON file and use this open source JSON parser. https://github.com/nlohmann/json?tab=readme-ov-file#examples

Reply:
Yes, I do indeed plan on doing that. I wanted to get the basic part of the C++ API done first before writing the JSON. I was probably gonna use JsonCPP, since I have used it before for work.

Reply:
The one you linked though looks more modern and updated, so I might use that instead.